### ISSUE TITLE
`module_name_repetitions` on exported macros should use crate name, not module name

### ISSUE THREAD

**Issue by bonzini** [rust-lang/rust-clippy#14095]
### Summary

Exported macros end up as symbols on the crate, leading to an almost-certain `module_name_repetitions` false positive.

### Lint Name

module_name_repetitions

### Reproducer

I tried this code:

```rust
pub mod widget {
    #[macro_export]
    macro_rules! define_widget {
        ($id:ident) => { /* ... */ }
    }
}
```

with this invocation of clippy-driver:
```shell
$ clippy-driver foo.rs -D clippy::module_name_repetitions --crate-type lib
```

I saw this happen:

```
error: item name ends with its containing module's name
 --> f.rs:3:18
  |
3 |     macro_rules! define_widget {
  |                  ^^^^^^^^^^^^^
  |
```

I expected to see this happen:

No warning. The macro is imported with `use foo::define_widget` rather than `use foo::widget::define_widget`, thus there's no repetition.

### Version

```text
rustc 1.84.0-nightly (bc5cf994d 2024-11-05)
binary: rustc
commit-hash: bc5cf994db9fb46712cefd89f78ad7fc51f184a2
commit-date: 2024-11-05
host: x86_64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.3
```
